addons/sys_intercom/fnc_updateInfantryPhoneStatus.sqf

```diff
@@ -6,38 +6,59 @@
+
+private "_intercomText";
+switch (_intercomNetwork) do {
+    case CREW_INTERCOM: {_interComText = "(" + localize LSTRING(crewIntercom) + ")";};
+    case PASSENGER_INTERCOM: {_interComText = "(" + localize LSTRING(passengerIntercom) + ")";};
+};
 
 switch (_action) do {
     case 0: {
         // Stop using the intercom externally
         _vehicle setVariable [QGVAR(unitInfantryPhone), nil, true];
         _unit setVariable [QGVAR(vehicleInfantryPhone), nil, true];
-        [localize LSTRING(infantryPhoneDisconnected), ICON_RADIO_CALL] call EFUNC(sys_core,displayNotification);
+        if (_intercomNetwork == PASSENGER_INTERCOM) then {
+            [_vehicle, _unit, 0] call FUNC(updatePassengerIntercomStatus);
+        };
+        [format [localize LSTRING(infantryPhoneDisconnected), _intercomText], ICON_RADIO_CALL] call EFUNC(sys_core,displayNotification);
     };
     case 1: {
-        _vehicle setVariable [QGVAR(unitInfantryPhone), _unit, true];
-        _unit setVariable [QGVAR(vehicleInfantryPhone), _vehicle, true];
-        [localize LSTRING(infantryPhoneConnected), ICON_RADIO_CALL] call EFUNC(sys_core,displayNotification);
+        // Start using the intercom externally
+        _vehicle setVariable [QGVAR(unitInfantryPhone), [_unit, _intercomNetwork], true];
+        _unit setVariable [QGVAR(vehicleInfantryPhone), [_vehicle, _intercomNetwork], true];
+        [format [localize LSTRING(infantryPhoneConnected), _intercomText], ICON_RADIO_CALL] call EFUNC(sys_core,displayNotification);
     };
     case 2: {
         // Give the intercom to another unit
-        _vehicle setVariable [QGVAR(unitInfantryPhone), _unit, true];
-        _unit setVariable [QGVAR(vehicleInfantryPhone), _vehicle, true];
+        _vehicle setVariable [QGVAR(unitInfantryPhone), [_unit, _intercomNetwork], true];
+        _unit setVariable [QGVAR(vehicleInfantryPhone), [_vehicle, _intercomNetwork], true];
         _givingUnit setVariable [QGVAR(vehicleInfantryPhone), nil, true];
-        [localize LSTRING(infantryPhoneReceived), ICON_RADIO_CALL, nil, _unit] call EFUNC(sys_core,displayNotification);
+        [format [localize LSTRING(infantryPhoneReceived), _intercomText], ICON_RADIO_CALL, nil, _unit] call EFUNC(sys_core,displayNotification);
+    };
+    case 3: {
+        // Switch to another intercom network
+        _vehicle setVariable [QGVAR(unitInfantryPhone), [_unit, _intercomNetwork], true];
+        _unit setVariable [QGVAR(vehicleInfantryPhone), [_vehicle, _intercomNetwork], true];
+        [format [localize LSTRING(infantryPhoneSwitched), _intercomText], ICON_RADIO_CALL] call EFUNC(sys_core,displayNotification);
     };
 };
+
+// Hook for third party mods with actions when picking returning infantry phone
+private _event = _vehicle getVariable [QGVAR(eventInfantryPhone), FUNC(noApiFunction)];
+[_vehicle, _unit, _action] call (missionNamespace getVariable [_event, FUNC(noApiFunction)]);
```

Comment:
If no event defined (set default value to `""`), don't do anything.

Also missing empty newline at end of file.